packages/protocol/contracts/governance/Election.sol

```diff
@@ -0,0 +1,779 @@
+    PendingVotes storage pending = votes.pending;
+    uint256 newValue = pending.balances[group][account].sub(value);
+    pending.balances[group][account] = newValue;
+    if (newValue == 0) {
+      pending.timestamps[group][account] = 0;
+    }
+    pending.total[group] = pending.total[group].sub(value);
+  }
+
+  /**
+   * @notice Increments the number of active votes for `group` made by `account`.
+   * @param group The address of the validator group.
+   * @param account The address of the voting account.
+   * @param value The number of votes.
+   */
+  function incrementActiveVotes(address group, address account, uint256 value) private {
+    uint256 delta = getActiveVotesDelta(group, value);
+    ActiveVotes storage active = votes.active;
+    active.numerators[group][account] = active.numerators[group][account].add(delta);
+    active.denominators[group] = active.denominators[group].add(delta);
+    active.total[group] = active.total[group].add(value);
+  }
+
+  /**
+   * @notice Decrements the number of active votes for `group` made by `account`.
+   * @param group The address of the validator group.
+   * @param account The address of the voting account.
+   * @param value The number of votes.
+   */
+  function decrementActiveVotes(address group, address account, uint256 value) private {
+    uint256 delta = getActiveVotesDelta(group, value);
+    ActiveVotes storage active = votes.active;
+    active.numerators[group][account] = active.numerators[group][account].sub(delta);
+    active.denominators[group] = active.denominators[group].sub(delta);
+    active.total[group] = active.total[group].sub(value);
+  }
+
+  /**
+   * @notice Returns the delta in active vote denominator for `group`.
+   * @param group The address of the validator group.
+   * @param value The numebr of active votes being added.
+   * @return The delta in active vote denominator for `group`.
+   */
+  function getActiveVotesDelta(address group, uint256 value) private view returns (uint256) {
+    // Preserve delta * total = value * denominator
+    return value.mul(votes.active.denominators[group].add(1)).div(
+      votes.active.total[group].add(1)
+    );
+  }
+
+  function getGroupsVotedFor(address account) external view returns (address[] memory) {
+    return votes.groupsVotedFor[account];
+  }
+
+  /**
+   * @notice Deletes an element from a list of addresses.
+   * @param list The list of addresses.
+   * @param element The address to delete.
+   * @param index The index of `element` in the list.
+   */
+  function deleteElement(address[] storage list, address element, uint256 index) private {
+    require(index < list.length && list[index] == element);
+    uint256 lastIndex = list.length.sub(1);
+    list[index] = list[lastIndex];
+    list[lastIndex] = address(0);
+    list.length = lastIndex;
+  }
+
+  /**
+   * @notice Returns the number of votes that a group can receive.
```

Comment:
Nice.

Still, it would be nice to understand what happens if after the method is called, some of the other variables change? (totalLockedGold, #groupMembers, #maxElectableValidators, #registeredValidators). That would probably break the the invariant; and i don't know if the rest of the code relies on it.